mesonbuild/compilers/mixins/gnu.py

```diff
@@ -397,3 +397,6 @@ def use_linker_args(cls, linker: str, version: str) -> T.List[str]:
         if linker == 'mold' and mesonlib.version_compare(version, '>=12.0.1'):
             return ['-fuse-ld=mold']
         return super().use_linker_args(linker, version)
+
+    def analyzer_compile_args(self) -> T.List[str]:
```

Comment:
Seems like you need to protect this with a version check of the compiler. Same thing for clang. Otherwise this looks like it would always return the argument as supported even if it is not.